Modify opent2t library to be instance based & added telelmetry tracking

ChuckFerring-zz opened the pull request:
Modified the opent2t library to be instance-based.
Added telemetry tracking to logger.
Added telemetry events and exceptions to opent2t library.

This is a breaking change.  I have PR's ready to go for the other repos but this change needs to go in first.

chuckferring committed:
Initial telemetry implementation.

chuckferring committed:
updates

chuckferring committed:
Updates.

chuckferring committed:
Removed unneeded proprty from ITrackerTransports and changed the default log level to verbose.

chuckferring committed:
Adding translator name to telemetry data.

arjun-msft commented on node/lib/Logger.ts:
```diff
@@ -1,58 +1,55 @@
+    Error = 3
+}
+
+export class Logger {
```
add a class level comment at the least #Closed

arjun-msft commented on node/lib/Logger.ts:
```diff
@@ -72,15 +69,43 @@ export class Logger implements ILogger {
+        return this.logger._names;
+    }
+
+    public normalizeWith(normalizer: (logObject: any) => any): Logger {
```
where is this used? #Closed

ChuckFerring-zz replied:
Doesn't look like it was used anywhere, removed.

---
In reply to: [109506842](https://github.com/openT2T/opent2t/pull/40#discussion_r109506842) [](ancestors = 109506842)

arjun-msft commented on node/lib/OpenT2T.ts:
```diff
@@ -128,20 +141,23 @@ export class OpenT2T {
+        let trackingData: any = { propertyName, schemaName };
+        let translatorForSchema = this.getTranslatorForSchema(translator, schemaName);
+        this.validateMemberName(propertyName);
+        let memberName = this.uncapitalize(propertyName);
```
break up this section with some whitespace #Closed

arjun-msft commented on node/lib/OpenT2T.ts:
```diff
@@ -214,6 +235,8 @@ export class OpenT2T {
             await result;
         }
+
+        this.logger.event("SetProperty", Date.now() - startTime, trackingData);
```
this goes for all methods, you should log an event in the error cases as well #Closed

arjun-msft commented on node/lib/OpenT2T.ts:
```diff
@@ -178,32 +197,34 @@ export class OpenT2T {
             `setPropertyAsync for : '${propertyName}' for translator schema: ${schemaName}`);
-        let translatorForSchema = OpenT2T.getTranslatorForSchema(translator, schemaName);
-        OpenT2T.validateMemberName(propertyName);
+        let trackingData: any = { propertyName, schemaName, value };
```
this goes for all cases where translator is passed in, log translator name #Closed

arjun-msft commented:
:shipit:

arjun-msft commented on node/lib/AppInsightsTrackerTransport.ts:
```diff
@@ -0,0 +1,53 @@
+    private client: any;
+
+    constructor(key: string) {
+        this.client = applicationinsights.getClient(key);
```
add session id #Closed

chuckferring committed:
Updates for PR feedback.

	Logging translator names in all calls.
	Added comments for Logger class.
	Removed unused methods in Logger class.
	Added error events.
	Added session Id for app insights telemetry tracker.

chuckferring committed:
Changed the order of parameters passed to translator constructors.

chuckferring committed:
Updated library version to 1.1.0.
